lte/cloud/go/services/subscriberdb/obsidian/handlers/handlers.go

```diff
@@ -862,6 +893,9 @@ func loadAllStatesForIMSIs(networkID string, imsis []string) (map[string]state_t
 				glog.Errorf("mobilityd state composite ID %s did not match regex", sidKey)
 				continue
 			}
+			if _, exists := imsiMap[matches[1]]; !exists && len(imsiMap) != 0 {
```

Comment:
More of a question then a review comment :) 
what is the clause `len(imsiMap) != 0` for?

Reply:
Hi Marie! Here's my thought process: the function `loadAllStatesForIMSIs` is used with the `imsis` field empty in `listSubscriberStateHandler`, in which case the expected behavior is to load the states for all subscribers, regardless of imsi. Therefore, when the `imsis` arg is an empty string, there's no need for `loadAllStatesForIMSIs` to filter out states of subscribers whose keys are not contained in `imsis` (which is empty). @themarwhal

Reply:
So yes, basically the `imsis` arg to this fn can be empty, in which case we search for "all" IMSIs. Let's handle this a little cleaner though

1. Document that functionality in the function docstring
2. Rather than "empty", indicate `imsis` should be `nil` to search all
3. Create an inline helper function, e.g. `shouldLoadIMSI`, that encapsulates this line (the map check and whether `imsis` is nil)